gazebo_color_plugin/src/color_plugin.cpp

```diff
@@ -0,0 +1,128 @@
+#include <gazebo/common/Plugin.hh>
+#include <ros/ros.h>
+#include <ros/console.h>
+#include "gazebo_color_sensor_plugin/color_sensor_plugin.h"
+
+#include "gazebo_plugins/gazebo_ros_camera.h"
+
+#include <string>
+#include <stdlib.h>
+
+#include <gazebo/sensors/Sensor.hh>
+#include <gazebo/sensors/CameraSensor.hh>
+#include <gazebo/sensors/SensorTypes.hh>
+#include <std_msgs/Bool.h>
+
+#include <sensor_msgs/Illuminance.h>
+
+#define COLOR_TOLERANCE 30
+
+namespace gazebo
+{
+  // Register this plugin with the simulator
+  GZ_REGISTER_SENSOR_PLUGIN(GazeboRosColor)
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Constructor
+  GazeboRosColor::GazeboRosColor():
+  _nh("color_sensor_plugin"),
+  _fov(6),
+  _range(10),
+  CameraPlugin(),
+  GazeboRosCameraUtils()
+
+  {
+    
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Destructor
+  GazeboRosColor::~GazeboRosColor()
+  {
+    ROS_DEBUG_STREAM_NAMED("camera","Unloaded");
+  }
+
+  void GazeboRosColor::Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf)
+  {
+    // Make sure the ROS node for Gazebo has already been initialized
+    if (!ros::isInitialized())
+    {
+      ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load plugin. "
+        << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
+      return;
+    }
+
+    CameraPlugin::Load(_parent, _sdf);
+    // copying from CameraPlugin into GazeboRosCameraUtils
+    GazeboRosCameraUtils::Load(_parent, _sdf);
+    this->parentSensor_ = this->parentSensor;
+    this->width_ = this->width;
+    this->height_ = this->height;
+    this->depth_ = this->depth;
+    this->format_ = this->format;
+    this->camera_ = this->camera;
+    this->publish_topic_name_ = _sdf-> Get<std::string>("publishTopic");
+    this->sensor_color_ = _sdf-> Get<std::string>("sensorColor");
+    this->update_rate_ = atof(_sdf-> Get<std::string>("update_rate").c_str());
```

Comment:
I'd suggest using consistent naming for tags/attributes (publishTopic, sensorColor and update_rate). 
If you use camelCase for one tag, use it for the rest of the tags. Anyway, I'd check if there is a convention for these kind of tags.

Reply:
Adopted lower_case naming